lib/deployment-bucket-endpoint.js

```diff
@@ -0,0 +1,14 @@
+'use-strict';
+
+const { endpoint } = require('aws-info');
+
+module.exports = function setDeploymentBucketEndpoint() {
+  const bucket = this.serverless.service.provider.deploymentBucket;
+  if ( bucket !== undefined ) {
+    return this.provider.request('S3', 'getBucketLocation', { Bucket: bucket }).then((response) => {
+       this.deploymentBucketEndpoint = endpoint('S3', response.LocationConstraint);
+    })
+  } else {
+     this.deploymentBucketEndpoint = endpoint('S3', this.options.region || this.serverless.service.provider.region);
```

Comment:
do you know if this handles things like AWS_REGION being set?

Reply:
Yes it ignores them. I tried with `AWS_REGION` and `AWS_DEFAULT_REGION`